File: prime/prime/page/hospital_overview/hospital_overview.js
frappe.pages["hospital-overview"].on_page_load = function (wrapper) {
	var page = frappe.ui.make_app_page({
		parent: wrapper,
		title: "Hospital Overview",
		single_column: true,
	});

	page.add_action_icon("refresh", function () { load(); });

	page.body.html(`
		<div class="dash-page">
			<div class="dash-header">
				<div>
					<div class="dash-title">Hospital Overview</div>
					<div class="dash-subtitle">Real-time overview of hospital operations</div>
				</div>
				<div class="dash-refresh-info" id="ho-last-refresh"></div>
			</div>
			<div id="ho-kpis" class="dash-kpi-grid">
				<div class="dash-loading">Loading...</div>
			</div>
			<div class="dash-chart-row two-col">
				<div class="dash-chart-card">
					<div class="dash-chart-title">Revenue Overview</div>
					<div class="dash-chart-heading">Monthly Revenue (6 months)</div>
					<div class="dash-chart-wrap" id="ho-chart-revenue"></div>
				</div>
				<div class="dash-chart-card">
					<div class="dash-chart-title">Today</div>
					<div class="dash-chart-heading">Appointments by Department</div>
					<div class="dash-chart-wrap" id="ho-chart-dept"></div>
				</div>
			</div>
			<div class="dash-chart-row two-col">
				<div class="dash-chart-card">
					<div class="dash-chart-title">This Month</div>
					<div class="dash-chart-heading">Top Doctors by Appointments</div>
					<table class="dash-table" id="ho-top-doctors">
						<thead><tr><th>Doctor</th><th>Appointments</th><th>Seen</th></tr></thead>
						<tbody></tbody>
					</table>
				</div>
				<div class="dash-chart-card">
					<div class="dash-chart-title">Today</div>
					<div class="dash-chart-heading">Queue Status</div>
					<div class="dash-chart-wrap" id="ho-chart-queue"></div>
				</div>
			</div>
		</div>
	`);

	function fmt_currency(v) {
		return "SOS " + parseFloat(v || 0).toLocaleString("en-US", { minimumFractionDigits: 0, maximumFractionDigits: 0 });
	}

	function load() {
		frappe.call({
			method: "prime.prime.page.hospital_overview.hospital_overview_api.get_stats",
			callback: function (r) {
				if (!r.message) return;
				var d = r.message;
				render_kpis(d.kpis);
				render_chart("ho-chart-revenue", "bar", d.monthly_revenue, ["#2563eb"], false);
				render_chart("ho-chart-dept", "pie", d.by_department, ["#2563eb","#10b981","#f59e0b","#ef4444","#8b5cf6","#06b6d4","#f97316","#84cc16"], true);
				render_chart("ho-chart-queue", "pie", d.queue_status, ["#f59e0b","#2563eb","#10b981","#ef4444","#6b7280"], true);
				render_top_doctors(d.top_doctors);
				document.getElementById("ho-last-refresh").textContent = "Updated " + frappe.datetime.now_time();
			}
		});
	}

	function render_kpis(k) {
		var cards = [
			{ icon: "🏥", color: "blue",   value: k.patients_today,    label: "Appointments Today" },
			{ icon: "📅", color: "purple", value: k.patients_month,    label: "Appointments This Month" },
			{ icon: "💰", color: "green",  value: fmt_currency(k.revenue_today),  label: "Revenue Today" },
			{ icon: "📊", color: "teal",   value: fmt_currency(k.revenue_month),  label: "Revenue This Month" },
			{ icon: "⏳", color: "amber",  value: k.queue_active,      label: "Active Queue" },
			{ icon: "👤", color: "indigo", value: k.new_patients,      label: "New Patients Today" },
			{ icon: "🧾", color: "red",    value: k.pending_invoices,  label: "Pending Invoices" },
			{ icon: "💳", color: "pink",   value: fmt_currency(k.outstanding_amount), label: "Outstanding Amount" },
		];
		document.getElementById("ho-kpis").innerHTML = cards.map(function (c) {
			return '<div class="dash-kpi-card">' +
				'<div class="dash-kpi-icon ' + c.color + '">' + c.icon + '</div>' +
				'<div><div class="dash-kpi-value">' + c.value + '</div><div class="dash-kpi-label">' + c.label + '</div></div>' +
				'</div>';
		}).join("");
	}

	function render_chart(id, type, data, colors, is_pie) {
		var el = document.getElementById(id);
		if (!el || !data || !data.length) { el && (el.innerHTML = '<div class="dash-loading">No data</div>'); return; }
		el.innerHTML = "";
		new frappe.Chart(el, {
			type: type,
			data: {
				labels: data.map(function (r) { return r.label; }),
				datasets: [{ values: data.map(function (r) { return r.value || 0; }) }]
			},
			height: 220,
			colors: colors,
			truncateLegends: true,
			tooltipOptions: { formatTooltipY: function (v) { return v; } }
		});
	}

	function render_top_doctors(rows) {
		var tbody = document.querySelector("#ho-top-doctors tbody");
		if (!tbody) return;
		tbody.innerHTML = (rows || []).map(function (r) {
			return "<tr><td>" + (r.doctor || "-") + "</td><td>" + r.patients + "</td><td>" + r.seen + "</td></tr>";
		}).join("") || '<tr><td colspan="3" style="text-align:center;color:#bbb">No data</td></tr>';
	}

	load();
	setInterval(load, 60000);
};

File: prime/prime/page/hospital_overview/hospital_overview.py
import frappe
from frappe.utils import today, get_first_day, flt


@frappe.whitelist()
def get_stats():
    td = today()
    first_day = str(get_first_day(td))

    def q(sql, *args):
        return frappe.db.sql(sql, args or None, as_dict=True)

    def scalar(sql, *args):
        r = frappe.db.sql(sql, args or None)
        return flt(r[0][0]) if r else 0

    # ── KPIs ──────────────────────────────────────────────────────────────────
    patients_today = scalar(
        "SELECT COUNT(*) FROM `tabPatient Appointment` WHERE appointment_date=%s AND status!='Cancelled'", td)
    patients_month = scalar(
        "SELECT COUNT(*) FROM `tabPatient Appointment` WHERE appointment_date>=%s AND status!='Cancelled'", first_day)
    revenue_today = scalar(
        "SELECT COALESCE(SUM(grand_total),0) FROM `tabSales Invoice` WHERE DATE(posting_date)=%s AND docstatus=1", td)
    revenue_month = scalar(
        "SELECT COALESCE(SUM(grand_total),0) FROM `tabSales Invoice` WHERE posting_date>=%s AND docstatus=1", first_day)
    pending_invoices = scalar(
        "SELECT COUNT(*) FROM `tabSales Invoice` WHERE outstanding_amount>0 AND docstatus=1")
    outstanding_amount = scalar(
        "SELECT COALESCE(SUM(outstanding_amount),0) FROM `tabSales Invoice` WHERE outstanding_amount>0 AND docstatus=1")
    queue_active = scalar(
        "SELECT COUNT(*) FROM `tabQue` WHERE date=%s AND status NOT IN ('Completed','Cancelled')", td)
    new_patients = scalar(
        "SELECT COUNT(*) FROM `tabPatient` WHERE DATE(creation)=%s", td)

    # ── Monthly revenue (last 6 months) ──────────────────────────────────────
    monthly_raw = frappe.db.sql("""
        SELECT YEAR(posting_date) AS yr, MONTH(posting_date) AS mo,
               COALESCE(SUM(grand_total),0) AS value
        FROM `tabSales Invoice`
        WHERE docstatus=1 AND posting_date >= DATE_SUB(CURDATE(), INTERVAL 6 MONTH)
        GROUP BY YEAR(posting_date), MONTH(posting_date)
        ORDER BY YEAR(posting_date), MONTH(posting_date)
    """, as_dict=True)
    import calendar
    monthly = [{"label": calendar.month_abbr[int(r.mo)] + " " + str(r.yr), "value": r.value} for r in monthly_raw]

    # ── Appointments by department today ─────────────────────────────────────
    by_dept = q("""
        SELECT COALESCE(department,'General') AS label, COUNT(*) AS value
        FROM `tabPatient Appointment`
        WHERE appointment_date=%s AND status!='Cancelled'
        GROUP BY department
        ORDER BY value DESC LIMIT 8
    """, td)

    # ── Top doctors this month ────────────────────────────────────────────────
    top_doctors = q("""
        SELECT practitioner_name AS doctor, COUNT(*) AS patients,
               SUM(CASE WHEN status='Closed' THEN 1 ELSE 0 END) AS seen
        FROM `tabPatient Appointment`
        WHERE appointment_date>=%s AND status!='Cancelled'
        GROUP BY practitioner
        ORDER BY patients DESC LIMIT 6
    """, first_day)

    # ── Queue by status today ─────────────────────────────────────────────────
    queue_status = q("""
        SELECT COALESCE(status,'Unknown') AS label, COUNT(*) AS value
        FROM `tabQue` WHERE date=%s
        GROUP BY status
    """, td)

    return {
        "kpis": {
            "patients_today": int(patients_today),
            "patients_month": int(patients_month),
            "revenue_today": revenue_today,
            "revenue_month": revenue_month,
            "pending_invoices": int(pending_invoices),
            "outstanding_amount": outstanding_amount,
            "queue_active": int(queue_active),
            "new_patients": int(new_patients),
        },
        "monthly_revenue": monthly,
        "by_department": by_dept,
        "top_doctors": top_doctors,
        "queue_status": queue_status,
    }
